test/e2e/suites/unmanaged/unmanaged_functional_test.go

```diff
@@ -528,4 +528,142 @@ var _ = ginkgo.Context("[unmanaged] [functional]", func() {
 			Expect(len(controlPlaneMachines)).To(Equal(1))
 		})
 	})
+
+	ginkgo.Describe("Externally managed security groups", func() {
+		ginkgo.It("should create a cluster using external security groups", func() {
+			specName := "functional-test-external-securitygroups"
+			requiredResources = &shared.TestResource{EC2Normal: 2 * e2eCtx.Settings.InstanceVCPU, IGW: 1, NGW: 1, VPC: 1, ClassicLB: 1, EIP: 3}
+			requiredResources.WriteRequestedResources(e2eCtx, specName)
+			Expect(shared.AcquireResources(requiredResources, config.GinkgoConfig.ParallelNode, flock.New(shared.ResourceQuotaFilePath))).To(Succeed())
+			defer shared.ReleaseResources(requiredResources, config.GinkgoConfig.ParallelNode, flock.New(shared.ResourceQuotaFilePath))
+			namespace := shared.SetupSpecNamespace(ctx, specName, e2eCtx)
+			defer shared.DumpSpecResourcesAndCleanup(ctx, "", namespace, e2eCtx)
+
+			ginkgo.By("Creating the VPC and subnets")
+			clusterName := fmt.Sprintf("cluster-%s", util.RandomString(6))
+			vpc, vpcErr := shared.CreateVPC(e2eCtx, clusterName+"-vpc", "10.0.0.0/22")
+			Expect(vpcErr).To(BeNil())
+			vpcID := *vpc.VpcId
+			shared.SetEnvVar("VPC_ID", vpcID, false)
```

Comment:
Tests are running in parallel, if multiple tests use same env vars, it might cause issues.
As we are only using these particular variables in this test, this is fine for now. But a warning note would be good not to forget we cannot reuse these.

Reply:
I'll need to consider this further as I'm using them in the internal ELB and peering tests as well. Thanks for the heads up.